### ISSUE TITLE
Questions about evaluation

### ISSUE THREAD

**Issue by congxin0920** [clinplayer/Point2Skeleton#17]
Hello, I'm sorry to bother you again. I used the method you mentioned above to evaluate the pre-trained model you gave, but the effect is worse than that in your paper. Could you please elaborate on the evaluation method?

**Comment by clinplayer** [comment]
I'm not sure what you exactly mean by "effect is worse". Are the numbers in a comparable scale with the paper? Do you normalize these numbers based on the sample numbers, i.e., 1/M\*Points2Spheres + 1/N\*Spheres2Points?  Note due to the random sampling, the numbers cannot be exactly the same.

**Comment by congxin0920** [comment]
My evaluation results are:  Average	0.0429	0.1339	0.1126	0.2028
Are the results normal?
I normalized these numbers based on the sample numbers, i.e., 1/M*Points2Spheres + 1/N*Spheres2Points.

**Comment by clinplayer** [comment]
Generally they look reasonable. Some numbers are better than the reported and some are worse. 
And one more thing, see L62-63 in test.py, please use ``util.rand_sample_points_on_skeleton_mesh`` to densely sample on the skeletal mesh, rather than directly using the predicted skeletal points.

**Comment by congxin0920** [comment]
To compute CD-recon and HD-recon ,Whether the input point cloud   is a dense point cloud or a point cloud after sampling？

**Comment by congxin0920** [comment]
Hi, I'm sorry to bother you again. Whether the average evaluation results(0.0372,0.1424,0.08828,0.1898) in your paper is obtained by dividing the sum of the results of each category by the number of categories.

**Comment by clinplayer** [comment]
Whether the input point cloud is a dense point cloud or a point cloud after sampling？
-- Do you mean groudtruth? The groundtruth point clouds for evaluation are dense ones, which are given in the README.
Whether the average evaluation results(0.0372,0.1424,0.08828,0.1898) in your paper is obtained by dividing the sum of the results of each category by the number of categories.
-- No, it is obtained by the mean of all the individual shapes.

**Comment by congxin0920** [comment]
> I'm not sure what you exactly mean by "effect is worse". Are the numbers in a comparable scale with the paper? Do you normalize these numbers based on the sample numbers, i.e., 1/M*Points2Spheres + 1/N*Spheres2Points? Note due to the random sampling, the numbers cannot be exactly the same.


Sorry I also have a little problem, What is the M value in the above formula（1/M*Points2Spheres + 1/N*Spheres2Points） when evaluating CD-RECON?

**Comment by clinplayer** [comment]
> > I'm not sure what you exactly mean by "effect is worse". Are the numbers in a comparable scale with the paper? Do you normalize these numbers based on the sample numbers, i.e., 1/M_Points2Spheres + 1/N_Spheres2Points? Note due to the random sampling, the numbers cannot be exactly the same.
> 
> Sorry I also have a little problem, What is the M value in the above formula（1/M_Points2Spheres + 1/N_Spheres2Points） when evaluating CD-RECON?

They are just parameters for averaging. M is the number of points of a groundtruth point cloud, and N is the number of skeletal spheres.
